algorithm_services/app.py

```diff
@@ -1,20 +1,48 @@
 import json
+import markdown
+import re
+import os.path
 
 from flask import Flask
 from flask import render_template
 from flask import Response
+from flask import Markup
 
 from algorithm_services.algorithms.fizzbuzz import fizzbuzz
 from algorithm_services.algorithms.clock_angle import clock_angle
 
 
+def get_index_data(readme_data):
+    found = re.search(
+        '### Available algorithms(.+?)### Tests',
+        readme_data,
+        re.DOTALL
+    )
+    if found:
+        return found.group(1)
+    else:
+        return ''
```

Comment:
Remember, you always need to test every condition in you logic. In this case, you have only tested if the regex found something. Test all behaviours, ok?